fuel-client/src/client.rs

```diff
@@ -264,6 +267,31 @@ impl FuelClient {
         let coins = self.query(query).await?.coins.into();
         Ok(coins)
     }
+
+    /// Retrieve coins to spend in a transaction
+    pub async fn coins_to_spend(
+        &self,
+        owner: &str,
+        spend_query: Vec<(&str, u64)>,
+        max_inputs: Option<usize>,
+    ) -> io::Result<Vec<schema::coin::Coin>> {
+        let owner: HexString256 = owner.parse()?;
+        let spend_query: Vec<SpendQueryElementInput> = spend_query
+            .iter()
+            .map(|(color, amount)| -> Result<_, ConversionError> {
+                Ok(SpendQueryElementInput {
+                    color: color.parse()?,
+                    amount: (*amount).into(),
+                })
+            })
+            .try_collect()?;
+        let max_inputs: Option<i32> = max_inputs.map(|max_inputs| max_inputs as i32);
```

Comment:
should we check if max_inputs is > `u32::MAX`? or the limit enforced somewhere else? otherwise a potential data loss is possible (usize on 64bit machine -> i32).